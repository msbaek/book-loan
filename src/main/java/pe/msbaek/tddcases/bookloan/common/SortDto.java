package pe.msbaek.tddcases.bookloan.common;

public record SortDto(String by, String direction) {
}
