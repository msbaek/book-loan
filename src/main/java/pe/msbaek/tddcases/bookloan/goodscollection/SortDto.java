package pe.msbaek.tddcases.bookloan.goodscollection;

public record SortDto(String by, String direction) {
}
