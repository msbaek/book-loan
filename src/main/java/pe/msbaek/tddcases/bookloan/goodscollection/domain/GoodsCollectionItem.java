package pe.msbaek.tddcases.bookloan.goodscollection.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Comment;

@NoArgsConstructor
@Getter
@Entity
@Table(name = "GOODS_COLLECTION_ITEM")
@Comment("특정 상품군 아이템")
public class GoodsCollectionItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Comment("특정 상품군 아이템 ID")
    private Long id;
    @Column(name = "GOODS_NO")
    @Comment("상품 No")
    private Long goodsNo;
    @Column(name = "GOODS_ID")
    @Comment("상품 ID")
    private String goodsId;
    @Column(name = "BARCODE")
    @Comment("상품 바코드")
    private String barcode;
    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "GOODS_COLLECTION_ID", nullable = false)
    private GoodsCollection goodsCollection;
    public static GoodsCollectionItem of(Goods goods) {
        GoodsCollectionItem item = new GoodsCollectionItem();
        item.goodsNo = goods.goodsNo();
        item.goodsId = goods.goodsId();
        item.barcode = goods.barcode();
        return item;
    }

    public void setId(Long id) {
        this.id = id;
    }
}