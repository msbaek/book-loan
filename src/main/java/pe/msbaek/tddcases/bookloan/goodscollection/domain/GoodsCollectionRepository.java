package pe.msbaek.tddcases.bookloan.goodscollection.domain;

import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface GoodsCollectionRepository {
    List<GoodsCollection> findByNameContaining(String name, Pageable pageable);
    Optional<GoodsCollection> findById(Long id);

    List<Goods> listGoodsByIds(List<String> ids);

    GoodsCollection save(GoodsCollection goodsCollection);
}