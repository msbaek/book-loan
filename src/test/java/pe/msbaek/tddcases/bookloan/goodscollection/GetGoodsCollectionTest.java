package pe.msbaek.tddcases.bookloan.goodscollection;

import com.ktown4u.utils.YamlPrinter;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.springframework.data.domain.Page;
import pe.msbaek.tddcases.bookloan.common.SearchDto;
import pe.msbaek.tddcases.bookloan.common.SortDto;

import static org.junit.jupiter.api.Assertions.*;

class GetGoodsCollectionTest {
    private GetGoodsCollection getGoodsCollection = new GetGoodsCollection();

    @Test
    void pagedGoodsCollection() {
        SearchDto request = new SearchDto("type", "keyword", 0, 10, new SortDto("by", "asc"));
        Page<GetGoodsCollection.GoodsCollectionDto> goodsCollectionDtos = getGoodsCollection.pagedGoodsCollection(request);
        Approvals.verify(YamlPrinter.print(goodsCollectionDtos.getContent()));
    }
}