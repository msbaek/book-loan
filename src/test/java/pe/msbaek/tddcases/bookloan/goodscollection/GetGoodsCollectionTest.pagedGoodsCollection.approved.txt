---
- id: 1
  name: "name1"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112216
    goodsId: "GD00112216"
    barcode: "9000000112216"
  - goodsNo: 112216
    goodsId: "GD00112216"
    barcode: "9000000112216"
  - goodsNo: 112216
    goodsId: "GD00112216"
    barcode: "9000000112216"
- id: 2
  name: "name2"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112215
    goodsId: "GD00112215"
    barcode: "9000000112215"
  - goodsNo: 112215
    goodsId: "GD00112215"
    barcode: "9000000112215"
  - goodsNo: 112215
    goodsId: "GD00112215"
    barcode: "9000000112215"
- id: 3
  name: "name3"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112214
    goodsId: "GD00112214"
    barcode: "9000000112214"
  - goodsNo: 112214
    goodsId: "GD00112214"
    barcode: "9000000112214"
  - goodsNo: 112214
    goodsId: "GD00112214"
    barcode: "9000000112214"
- id: 4
  name: "name4"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112213
    goodsId: "GD00112213"
    barcode: "8809868449639"
  - goodsNo: 112213
    goodsId: "GD00112213"
    barcode: "8809868449639"
  - goodsNo: 112213
    goodsId: "GD00112213"
    barcode: "8809868449639"
- id: 5
  name: "name5"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112212
    goodsId: "GD00112212"
    barcode: "8809868449639"
  - goodsNo: 112212
    goodsId: "GD00112212"
    barcode: "8809868449639"
  - goodsNo: 112212
    goodsId: "GD00112212"
    barcode: "8809868449639"
- id: 6
  name: "name6"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112210
    goodsId: "GD00112210"
    barcode: "8809315149921"
  - goodsNo: 112210
    goodsId: "GD00112210"
    barcode: "8809315149921"
  - goodsNo: 112210
    goodsId: "GD00112210"
    barcode: "8809315149921"
- id: 7
  name: "name7"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112208
    goodsId: "GD00112208"
    barcode: "8809888410275"
  - goodsNo: 112208
    goodsId: "GD00112208"
    barcode: "8809888410275"
  - goodsNo: 112208
    goodsId: "GD00112208"
    barcode: "8809888410275"
- id: 8
  name: "name8"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112207
    goodsId: "GD00112207"
    barcode: "8809888410275"
  - goodsNo: 112207
    goodsId: "GD00112207"
    barcode: "8809888410275"
  - goodsNo: 112207
    goodsId: "GD00112207"
    barcode: "8809888410275"
- id: 9
  name: "name9"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112206
    goodsId: "GD00112206"
    barcode: "9000000112206"
  - goodsNo: 112206
    goodsId: "GD00112206"
    barcode: "9000000112206"
  - goodsNo: 112206
    goodsId: "GD00112206"
    barcode: "9000000112206"
- id: 10
  name: "name10"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 112205
    goodsId: "GD00112205"
    barcode: "9000000112205"
  - goodsNo: 112205
    goodsId: "GD00112205"
    barcode: "9000000112205"
  - goodsNo: 112205
    goodsId: "GD00112205"
    barcode: "9000000112205"
